### ISSUE TITLE
Adding Grey Wolf Optimization Algorithm

### ISSUE THREAD

**Issue by NimishVerma** [HaaLeo/swarmlib#12]
Hi,
I checked out this project and it looks impressive! I have worked on EAs and SIs for some time and would like to contribute to this project by adding Grey Wolf Optimization Algorithm _(S. Mirjalili  et al 2014)_ .

**Comment by HaaLeo** [comment]
Hi @NimishVerma,
thank you for your interest in contributing to the `swarmlib` project 🚀 . I skimmed the [Grey Wolf Optimizer](https://www.sciencedirect.com/science/article/abs/pii/S0965997813001853) paper and it looks very interesting and suitable to be included in the `swarmlib`. 
I am happy to review a PR that implements the Grey Wolf Optimizer algorithm and its visualization similar to the existing algorithms (CS, PSO, ABC) by reusing the base components (like [`base_visualizer.py`](https://github.com/HaaLeo/swarmlib/blob/master/swarmlib/util/base_visualizer.py) or [`coordinate.py`](https://github.com/HaaLeo/swarmlib/blob/master/swarmlib/util/coordinate.py)). 
If you got any additional time, I would also greatly appreciate some first unit tests using `pytest` 😬 😉. 

To get started, I recommend you to take a look at the implemenation of the CS, PSO, ABC algorithms. The folder structure for the GWO algorithm should be very similar.
```
|
|- swarmlib
|    |- gwo
|    |   |- __init__.py
|    |   |- main.py                   // Register the argument parser for the GWO
|    |   |- gwo_problem.py     // Implement the GWO here similar to PSO etc. Main work will be done here.
|    |   |- wolf.py                    // Optional: Subclass of coordinate.py to add additional functions if needed
|    |   |- visualizer.py           // Optional: Subclass of base_visualizer.py if any additional functions are needed
|    |- __init__.py                   // import gwo_problem for easier API access
|    |- __main__.py                // Register GWO arg parser
|- swarmlib_test
|    |- gwo_test
|        |- ...
|- README.md                   // Extend readme with a brief introduction of the algorithm
```

If you face any issues I am happy to support.

**Comment by HaaLeo** [comment]
I derived some first [contribution guidelines](https://github.com/HaaLeo/swarmlib/blob/master/CONTRIBUTING.md#how-to-contribute) from my comment above.
I hope they help you getting started 🙂.

**Comment by NimishVerma** [comment]
Hi Leo,
Thanks for the guidelines, I will be contributing this soon.

**Comment by HaaLeo** [comment]
Fixed in #13 shipped with [v0.9.0](https://pypi.org/project/swarmlib/0.9.0/)
